content/deyemiobaa/building-a-todolist-with-rails.md

```diff
@@ -0,0 +1,351 @@
+
+## Setting up your dev environment: Installing Ruby, Rails, and PostgreSQL
+
+[GoRails](https://goRails.com/) Has in-depth guides on setting up Rails on various machines. Here's a link to all of them. Since we're using PostgreSQL for this project, skip any step that mentions setting up SQLite or MySQL.
+
+- [For Ubuntu](https://goRails.com/setup/ubuntu/22.04)
+- [Mac OS](https://goRails.com/setup/macos/13-ventura)
+- [For Windows](https://www.hanselman.com/blog/ruby-on-Rails-on-windows-is-not-just-possible-its-fabulous-using-wsl2-and-vs-code)
+
+## Setting up a new Rails app
+
+To create a new Rails app, simply open up your terminal, navigate to where you want this app to stay and run the following command.
+
+```bash
+rails new todolist -d postgresql –css tailwind
+```
+
+The commands after the app name (`todolist`) are optional. The `-d postgresql` flag tells rails to use PostgreSQL as the database for this app. The `–css tailwind` flag tells rails to use tailwind as the CSS framework for this app. You can use any CSS framework you want or just plain CSS. I chose tailwind because it’s easy to use and it’s a breeze to set up.
+
+After running the command, you’ll see a bunch of files and folders being created. This is the default structure of a Rails app. You can read more about it [here](https://guides.rubyonrails.org/getting_started.html#creating-the-blog-application).
+
+To get our app running, we need to make sure it is connected to a database. The database config lives in the `config/database.yml` file. There we can see the provided database name and some other info. This database doesn't exist yet and needs to be created. For most people, you need to supply your PostgreSQL username and password that was set during installation to this file. The default username is `postgres` and this applies to most Linux distros.
+
+Most times we would use an env file to manage this but we can keep it there for the scope of this project.
+
+```yml
+...
+default: &default
+  adapter: postgresql
+  encoding: unicode
+  username: postgres
+  password: yourpassword
+...
+```
+
+To create the database, run `rails db:create`.
+
+Now run `rails s` to start the server. If everything went well, you should see a url to visit (`http://127.0.0.1:3000/`) in your terminal. Open that url in your browser and you should see the default Rails page.
+
+Since we have TailwindCSS in our project, starting our server requires an additional step. We need to run `rails tailwindcss:watch` in a separate terminal window. This will compile our tailwind styles and make them available to our app.
+
+Rails ties both the server and the tailwind compiler together into a single command. To run both at the same time, run `bin/dev`.
```

Comment:
`bin\dev` does not work.

Reply:
What error does it give? If you're on windows you might get an error. A potential solution is to edit the Procfile.dev file.

Its contents look like this
```
web: bin/rails server -p 3000
css: bin/rails tailwindcss:watch
```

Update it to this.
```
web: rails s
css: rails tailwindcss:watch
```

We might just stick to running the commands in two terminals in none works.

Reply:
I'm running in Ubuntu. The error I get is: `bin/dev: 8: exec: foreman: not found`

Reply:
foreman should come bundled with ruby by default. Before you do run this `gem install foreman`. Run `bundle update` from the previous command and see if you're able to proceed.

Also it'll be helpful if I can have access to your source code for closer inspection

Reply:
ok installing foreman allowed `bin/dev` to work.

Reply:
So this has been reported as an [issue](https://github.com/rails/tailwindcss-rails#error-cannot-find-the-tailwindcss-executable-for-supported-platform) with the gems' support for some platforms.

I'll update the article with a manual setup for Tailwind.